AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -111,8 +111,15 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
     public func delete<M: Model>(_ modelType: M.Type,
                                  withId id: String,
                                  completion: @escaping DataStoreCallback<Void>) {
+        delete(modelType, modelSchema: modelType.schema, withId: id, completion: completion)
+    }
+    
+    public func delete<M: Model>(_ modelType: M.Type,
+                                 modelSchema: ModelSchema,
+                                 withId id: String,
+                                 completion: @escaping DataStoreCallback<Void>) {
         reinitStorageEngineIfNeeded()
-        storageEngine.delete(modelType, modelSchema: modelType.schema, withId: id) { result in
```

Comment:
Have you tried a global text search on `modelType.schema` to see if there are other places this needs to be updated?

Reply:
Yes, most of them are covered in this PR. There are 2-3 places related to subscription and usage of untyped model, which I will address in another PR.